server/handler/eval_context.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/palantir/policy-bot/policy"
+	"github.com/palantir/policy-bot/policy/common"
+	"github.com/palantir/policy-bot/pull"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/shurcooL/githubv4"
+)
+
+// EvalContext contains common fields and methods used to evaluate policy
+// requests. Handlers construct an EvalContext once they decide to handle a
+// request or event, then call the appropriate methods for each stage of
+// evaluation. Handlers with no special requirements can simply call Evaluate.
+type EvalContext struct {
```

Comment:
Thanks for doing this refactor. I think the EvalContext struct is a nice reorganization of the previous code.